pkg/compact/compact.go

```diff
@@ -41,6 +41,12 @@ const (
 	ResolutionLevel1h  = ResolutionLevel(downsample.ResLevel2)
 )
 
+const (
+	// DedupAlgorithmPenalty is the penalty based compactor series merge algorithm.
+	// This is the same as the online deduplication of querier except counter reset handling.
+	DedupAlgorithmPenalty = "penalty"
```

Comment:
Can we update https://thanos.io/tip/components/compact.md/#vertical-compaction-use-cases to name this algorithm `penalty`? We used to call it `realistic` (I think penalty based name is nice as it's very specific)

Reply:
Done.